transports/webrtc/src/tokio/upgrade/noise.rs

```diff
@@ -41,12 +42,11 @@ where
         .unwrap();
     let noise = NoiseConfig::xx(dh_keys)
         .with_prologue(noise_prologue(client_fingerprint, server_fingerprint));
-    let info = noise.protocol_info().next().unwrap();
     // Note the roles are reversed because it allows the server (webrtc connection responder) to
     // send application data 0.5 RTT earlier.
     let (peer_id, mut channel) = noise
         .into_authenticated()
-        .upgrade_outbound(stream, info)
+        .upgrade_outbound(stream, Protocol::from_static("/noise"))
```

Comment:
Mhhh, this is a code smell, i.e. `libp2p-webrtc` assumes that `libp2p-noise` does not make use of the `Protocol` and assumes that `libp2p-noise` does not change the `Protocol`. Can we do better? While not ideal, I prefer the previous solution, namely to call `ToProtocolIter` on `noise`.

Reply:
Really, whatever we pass here doesn't matter because the noise upgrade doesn't check this value. The actual problem here is that we are using noise on the wrong abstraction layer. We should be using an abstraction "below" the upgrade that doesn't care about the protocol string.

Reply:
But yes, we can revert this.